server/sonar-web/src/main/js/apps/portfolio/components/Activity.tsx

```diff
@@ -103,12 +105,13 @@ export default class Activity extends React.PureComponent<Props> {
         {this.state.loading ? (
           <i className="spinner" />
         ) : (
-          this.state.history &&
-          this.state.metrics && (
+          this.state.metrics != undefined &&
+          this.state.history != undefined && (
             <AnyPreviewGraph
               history={this.state.history}
               metrics={this.state.metrics}
               project={this.props.component}
+              renderWhenEmpty={this.renderWhenEmpty}
```

Comment:
I'd put directly the JSX object there, instead of using a function : 
```js
renderWhenEmpty={
   <div className="note">{translate('component_measures.no_history')}</div>
}
```

Reply:
I just wanted to avoid re-renders, but maybe it's not important here...

Reply:
Ah yeah true... I didn't think about it!